### PR TITLE
Adding new workflows

### PR DESCRIPTION
**Pull request by Gmerold**
Adds `promotion` and `CodeQL` workflows
Adds bug report template
Adds PR template

### CODE REVIEW

File: .github/workflows/promote.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            echo "promote-to=beta" >> ${GITHUB_ENV}
+          elif [ "${PROMOTE_FROM}" == "beta -> candidate" ]; then
+            echo "promote-from=beta" >> ${GITHUB_ENV}
+            echo "promote-to=candidate" >> ${GITHUB_ENV}
+          elif [ "${PROMOTE_FROM}" == "candidate -> stable" ]; then
+            echo "promote-from=candidate" >> ${GITHUB_ENV}
+            echo "promote-to=stable" >> ${GITHUB_ENV}
+          fi
+      - name: Promote Charm
+        uses: canonical/charming-actions/release-charm@2.2.2
```
Comment (sanchezfdezjavier) [inline .github/workflows/promote.yaml:36]: Do we want to freeze the version or use `latest` as you suggested?

File: .github/workflows/promote.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        with:
+          credentials: ${{ secrets.CHARMHUB_TOKEN }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          destination-channel: latest/${{ env.promote-to }}
+          origin-channel: latest/${{ env.promote-from }}
+          charmcraft-channel: latest/stable
```
Comment (saltiyazan) [inline .github/workflows/promote.yaml:42]: netpicking, but this is default :)

**Review by saltiyazan — APPROVED** [review]

File: .github/workflows/promote.yaml (hunk 2 above)
Comment (Gmerold) [reply .github/workflows/promote.yaml]: Right ;)
Removed.

File: .github/workflows/promote.yaml (hunk 1 above)
Comment (Gmerold) [reply .github/workflows/promote.yaml]: True.
Updated.

**Review by sanchezfdezjavier — APPROVED** [review]

**Pull request merged**